### PR TITLE
refactoring SolveSetup.py

### PR DESCRIPTION
**Pull request by maxcapodi78**

### CODE REVIEW

**Review by Samuelopez-ansys — APPROVED** [review]

File: examples/02-HFSS/Waveguide_Filter.py [hunk 1]
```diff
@@ -202,7 +202,7 @@ def place_iris(zpos, dz, n):
 #  Each frequency point is solved simultaneously.
 
 
-setup.analyze(num_tasks=2)
+setup.analyze(tasks=2)
 
 ###############################################################################
 # Generate S-Parameter Plots
```
Comment (PipKat) [inline examples/02-HFSS/Waveguide_Filter.py:208]:
```suggestion
# Generate S-parameter plots
```
Headings should be in sentence case.

File: examples/06-Multiphysics/MRI.py [hunk 2]
```diff
@@ -97,7 +97,7 @@
 # Save and analyze the project.
 
 hfss.save_project(os.path.join(project_path, "solved.aedt"))
-hfss.analyze(num_cores=6)
+hfss.analyze(cores=6)
 
 ###############################################################################
 # Plot SAR on Cut Plane in Phantom
```
Comment (PipKat) [inline examples/06-Multiphysics/MRI.py:103]:
```suggestion
# Plot SAR on cut plane in phantom
```

File: examples/06-Multiphysics/MRI.py [hunk 3]
```diff
@@ -283,7 +283,7 @@
 # Plot Temperature on cut plane.
 # Plot Temperature on monitor point.
```
Comment (PipKat) [inline examples/06-Multiphysics/MRI.py:284]:
```suggestion
# Plot temperature on cut plane.
# Plot temperature on monitor point.
```

File: examples/06-Multiphysics/MRI.py [hunk 4]
```diff
@@ -195,7 +195,7 @@
 # ~~~~~~~~~~~~~~~~~~
 # Analyze the project.
 
-mech.analyze(num_cores=6)
+mech.analyze(cores=6)
 
 ###############################################################################
 # Plot Fields
```
Comment (PipKat) [inline examples/06-Multiphysics/MRI.py:201]:
```suggestion
# Plot fields
```

File: pyaedt/application/Analysis.py [hunk 5]
```diff
@@ -1575,14 +1575,14 @@ def analyze(
 
         Parameters
         ----------
-        setup_name : str, optional
+        name : str, optional
             Setup to analyze. Default is ``None`` which solves all the setups.
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:1579]:
```suggestion
            Setup to analyze. The default is ``None``, in which case all
            setups are solved.
```

File: pyaedt/application/Analysis.py [hunk 6]
```diff
@@ -1575,14 +1575,14 @@ def analyze(
...
-        num_cores : int, optional
+        cores : int, optional
             Number of simulation cores. Default is ``4`` which is the number of cores available in license.
-        num_tasks : int, optional
+        tasks : int, optional
             Number of simulation tasks. Default is ``1``.
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:1583]:
```suggestion
            Number of simulation tasks. The default is ``1``.
```

File: pyaedt/application/Analysis.py [hunk 7]
```diff
@@ -1575,14 +1575,14 @@ def analyze(
...
             In bach solve, set num_tasks to ``-1`` to apply auto settings and distributed mode.
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:1584]:
```suggestion
            In bach solve, set ``num_tasks`` to ``-1`` to apply auto settings and distributed mode.
```

File: pyaedt/application/Analysis.py [hunk 8]
```diff
@@ -1575,14 +1575,14 @@ def analyze(
...
-        num_gpu : int, optional
+        gpus : int, optional
             Number of simulation graphic processing units to use. Default is ``0``.
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:1586]:
```suggestion
            Number of simulation graphic processing units to use. The default is ``0``.
```

File: pyaedt/application/Analysis.py [hunk 9]
```diff
@@ -1655,11 +1655,11 @@ def analyze_setup(
         name : str, optional
             Name of the setup, which can be an optimetric setup or a simple setup.
             If ``None`` all setups will be solved.
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:1657]:
```suggestion
            The default is ``None``, in which case all setups are solved.
```

File: pyaedt/application/Analysis.py [hunk 10]
```diff
@@ -1655,11 +1655,11 @@ def analyze_setup(
...
-        num_cores : int, optional
+        cores : int, optional
             Number of simulation cores.  Default is ``1``.
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:1659]:
```suggestion
            Number of simulation cores.  The default is ``1``.
```

File: pyaedt/application/Analysis.py [hunk 11]
```diff
@@ -1655,11 +1655,11 @@ def analyze_setup(
...
-        num_tasks : int, optional
+        tasks : int, optional
             Number of simulation tasks.  Default is ``1``.
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:1661]:
```suggestion
            Number of simulation tasks.  The default is ``1``.
```

File: pyaedt/application/Analysis.py [hunk 12]
```diff
@@ -1655,11 +1655,11 @@ def analyze_setup(
...
-        num_gpu : int, optional
+        gpus : int, optional
             Number of simulation graphics processing units.  Default is ``0``.
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:1663]:
```suggestion
            Number of simulation graphics processing units.  The default is ``0``.
```

File: pyaedt/application/Analysis.py [hunk 13]
```diff
@@ -1655,11 +1655,11 @@ def analyze_setup(
...
         acf_file : str, optional
             Full path to custom ACF file. The default is ``None.``
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:1665]:
```suggestion
            Full path to the custom ACF file. The default is ``None``.
```

File: pyaedt/application/Analysis.py [hunk 14]
```diff
@@ -1888,11 +1888,11 @@ def solve_in_batch(
         run_in_thread : bool, optional
             Whether to submit the batch command as a thread. The default is
             ``False``.
-        num_cores : int, optional
+        cores : int, optional
             Number of cores to use in simulation.
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:1892]:
```suggestion
            Number of cores to use in the simulation.
```

File: pyaedt/application/Analysis.py [hunk 15]
```diff
@@ -1888,11 +1888,11 @@ def solve_in_batch(
...
-        num_tasks : int, optional
+        tasks : int, optional
             Number of tasks to use in simulation. Set num_tasks to ``-1`` to apply auto settings and distributed mode.
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:1894]:
```suggestion
            Number of tasks to use in the simulation. Set ``num_tasks`` to ``-1`` to apply auto settings and distributed mode.
```

File: pyaedt/application/Analysis.py [hunk 16]
```diff
@@ -1888,11 +1888,11 @@ def solve_in_batch(
...
-        setup_name : str
+        setup : str
             Name of the setup, which can be an optimetric setup or a simple setup. If ``None`` all setup will be solved.
         revert_to_initial_mesh : bool, optional
             Whether to revert to initial mesh before solving or not. Default is ``False``.
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:1898]:
```suggestion
            Name of the setup, which can be an optimetric setup or a simple setup. The defaut is ``None``, in which case all setups are solved.
        revert_to_initial_mesh : bool, optional
            Whether to revert to the initial mesh before solving. The default is ``False``.
```

File: pyaedt/modules/DesignXPloration.py [hunk 17]
```diff
@@ -521,11 +521,11 @@ def analyze(
 
         Parameters
         ----------
-        num_cores : int, optional
+        cores : int, optional
             Number of simulation cores. Default is ``1``.
```
Comment (PipKat) [inline pyaedt/modules/DesignXPloration.py:525]:
```suggestion
            Number of simulation cores. The default is ``1``.
```

File: pyaedt/modules/DesignXPloration.py [hunk 18]
```diff
@@ -521,11 +521,11 @@ def analyze(
...
-        num_tasks : int, optional
+        tasks : int, optional
             Number of simulation tasks. Default is ``1``.
```
Comment (PipKat) [inline pyaedt/modules/DesignXPloration.py:527]:
```suggestion
            Number of simulation tasks. The default is ``1``.
```

File: pyaedt/modules/DesignXPloration.py [hunk 19]
```diff
@@ -521,11 +521,11 @@ def analyze(
...
-        num_gpu : int, optional
+        gpus : int, optional
             Number of simulation graphic processing units to use. Default is ``0``.
```
Comment (PipKat) [inline pyaedt/modules/DesignXPloration.py:529]:
```suggestion
            Number of simulation graphic processing units to use. The default is ``0``.
```

File: pyaedt/modules/SolveSetup.py [hunk 20]
```diff
@@ -88,11 +88,11 @@ def analyze(
 
         Parameters
         ----------
-        num_cores : int, optional
+        cores : int, optional
             Number of simulation cores. Default is ``1``.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:92]:
```suggestion
            Number of simulation cores. The default is ``1``.
```

File: pyaedt/modules/SolveSetup.py [hunk 21]
```diff
@@ -88,11 +88,11 @@ def analyze(
...
-        num_tasks : int, optional
+        tasks : int, optional
             Number of simulation tasks. Default is ``1``.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:94]:
```suggestion
            Number of simulation tasks. The default is ``1``.
```

File: pyaedt/modules/SolveSetup.py [hunk 22]
```diff
@@ -88,11 +88,11 @@ def analyze(
...
-        num_gpu : int, optional
+        gpus : int, optional
             Number of simulation graphic processing units to use. Default is ``0``.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:96]:
```suggestion
            Number of simulation graphic processing units to use. The default is ``0``.
```

File: pyaedt/modules/SolveSetup.py [hunk 23]
```diff
@@ -1193,9 +1193,9 @@ def add_sweep_count(
         ----------
         sweep_variable : str, optional
             Variable to which the sweep belongs. Default is ``"Freq``.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:1195]:
```suggestion
            Variable that the sweep belongs to. The default is ``"Freq``.
```

File: pyaedt/modules/SolveSetup.py [hunk 24]
```diff
@@ -1193,9 +1193,9 @@ def add_sweep_count(
...
-        start_point : float or str, optional
+        start : float or str, optional
             Start Point of Linear Count sweep. If ``str`` then no units will be applied.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:1197]:
```suggestion
            Start point of the linear count sweep. The default is ``1``.
            If a string ``str`` is specified, no units are applied.
```

File: pyaedt/modules/SolveSetup.py [hunk 25]
```diff
@@ -1193,9 +1193,9 @@ def add_sweep_count(
...
-        end_point : float or str, optional
+        stop : float or str, optional
             End Point of Linear Count sweep. If ``str`` then no units will be applied.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:1199]:
```suggestion
            End point of the linear count sweep. The default is ``100``.
            If a string is specified, no units are applied.
```

File: pyaedt/modules/SolveSetup.py [hunk 26]
```diff
@@ -1249,9 +1249,9 @@ def add_sweep_step(
         ----------
         sweep_variable : str, optional
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:1250]: Same suggestions as earlier, please.

File: pyaedt/modules/SolveSetup.py [hunk 27]
```diff
@@ -1249,9 +1249,9 @@ def add_sweep_step(
...
             Variable to which the sweep belongs. Default is ``"Freq``.
-        start_point : float or str, optional
+        start : float or str, optional
             Start Point of Linear Count sweep. If ``str`` then no units will be applied.
-        end_point : float or str, optional
+        stop : float or str, optional
             End Point of Linear Count sweep. If ``str`` then no units will be applied.
         step_size :  float or str, optional
             Step Size of sweep. If ``str`` then no units will be applied.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:1257]:
```suggestion
            Step size of the sweep. The default is ``1``.
            If a string is specified, no units are applied.
```

File: pyaedt/modules/SolveSetup.py [hunk 28]
```diff
@@ -1851,15 +1851,15 @@ def disable(self):
         self.update()
         return True
 
-    @pyaedt_function_handler()
-    def export_to_hfss(self, file_fullname, keep_net_name=False):
+    @pyaedt_function_handler(file_fullname="output_file")
+    def export_to_hfss(self, output_file, keep_net_name=False):
         """Export the HFSS 3D Layout design to HFSS 3D design.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:1856]: Should the concluding "Design" be capitalized, assuming it's part of an application name?

File: pyaedt/modules/SolveSetup.py [hunk 29]
```diff
@@ -2079,14 +2079,16 @@ def _check_export_log(self, info_messages, error_messages, file_fullname):
             time.sleep(2)
         return succeeded
 
-    @pyaedt_function_handler()
-    def export_to_q3d(self, file_fullname, keep_net_name=False):
+    @pyaedt_function_handler(file_fullname="output_file")
+    def export_to_q3d(self, output_file, keep_net_name=False):
         """Export the HFSS 3DLayout design to Q3D design.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:2084]:
```suggestion
        """Export the HFSS 3D Layout design to a Q3D design.
```

File: pyaedt/modules/SolveSetup.py [hunk 30]
```diff
@@ -2079,14 +2079,16 @@ def _check_export_log(self, info_messages, error_messages, file_fullname):
...
 
         Parameters
         ----------
-        file_fullname : str
+        output_file : str
             Full path and file name for exporting the project.
+        keep_net_name : bool
+            Keep net name in 3D export when ``True`` or by default when ``False``. Default value is ``False``.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:2091]:
```suggestion
            Whether to keep the net name in the 3D export, The default is ``False``.
```

File: pyaedt/application/Analysis.py [hunk 31]
```diff
@@ -1575,15 +1575,16 @@ def analyze(
...
+        cores : int, optional
             Number of simulation cores. Default is ``4`` which is the number of cores available in license.
-        num_tasks : int, optional
-            Number of simulation tasks. Default is ``1``.
-            In bach solve, set num_tasks to ``-1`` to apply auto settings and distributed mode.
-        num_gpu : int, optional
-            Number of simulation graphic processing units to use. Default is ``0``.
+        tasks : int, optional
+            Number of simulation tasks. The default is ``1``.
+            In bach solve, set ``num_tasks`` to ``-1`` to apply auto settings and distributed mode.
```
Comment (dcrawforAtAnsys) [inline pyaedt/application/Analysis.py:1585]:
```suggestion
            In bach solve, set ``tasks`` to ``-1`` to apply auto settings and distributed mode.
```

**Review by dcrawforAtAnsys — COMMENTED** [review]
LGTM.

**Review by Samuelopez-ansys — APPROVED** [review]

**Review by Samuelopez-ansys — APPROVED** [review]

**Pull request merged**